### PR TITLE
optimize Slice.opEquals

### PR DESCRIPTION
**Pull request by 9il**

### CODE REVIEW

File: std/experimental/ndslice/slice.d [hunk 1]
```diff
@@ -1737,17 +1738,27 @@ struct Slice(size_t _N, _Range)
...
+                return true;
+        import std.experimental.ndslice.selection : unpack;
+        return opEqualsImpl(this.unpack, rslice.unpack);
     }
 
     ///ditto
     bool opEquals(T)(T[] rarrary)
     {
-        if (this.length != rarrary.length)
+        auto slice = this;
```
Comment (JackStouffer) [inline std/experimental/ndslice/slice.d]: What's the reasoning behind this?

Comment (9il) [inline std/experimental/ndslice/slice.d]: `slice.popFront;` would change the slice

File: std/experimental/ndslice/slice.d [hunk 2]
```diff
@@ -1737,17 +1738,27 @@ struct Slice(size_t _N, _Range)
...
-        foreach (i, ref e; rarrary)
-            if (e != this[i])
+        if (rarrary.length) do
+        {
+            if (slice.front != rarrary.front)
                 return false;
+            slice.popFront;
+            rarrary.popFront;
+        }
+        while (rarrary.length);
```
Comment (Hackerpilot) [inline std/experimental/ndslice/slice.d]: I feel like this change is also part of one of the other pull requests.

Comment (9il) [inline std/experimental/ndslice/slice.d]: Oops, yes. Thank you! it is part of #4647

**Pull request closed without merging**